common/include/pcl/common/impl/bivariate_polynomial.hpp

```diff
@@ -289,9 +289,20 @@ pcl::BivariatePolynomialT<real>::readBinary (std::istream& os)
 {
   memoryCleanUp ();
   os.read (reinterpret_cast<char*> (&this->degree), sizeof (int));
-  unsigned int paramCnt = getNoOfParametersFromDegree (this->degree);
-  parameters = new real[paramCnt];
-  os.read (reinterpret_cast<char*> (&(*this->parameters)), paramCnt * sizeof (real));
+  if (os.bad())
+  {
+    PCL_THROW_EXCEPTION (pcl::IOException, "Failure in reading degree from file");
+  }
+  else if (os.fail())
+  {
+    PCL_THROW_EXCEPTION (pcl::IOException, "failbit set while reading degree (formatting or extraction error");
```

Comment:
The error message doesn't need to be technical. Something similar to "Bad formatting or input size detected" might work.